api/models/ml_model.py

```diff
@@ -0,0 +1,133 @@
+from pydantic import Field, validator, root_validator
+from typing import List, Optional
+
+from .base import SnakeModel
+
+
+class MLModelDTO(SnakeModel):
+    device: str = Field(example="Jetson")
+    name: str = Field(example="openvino")
+    image_size: str = Field(example="300,300,3")
+    model_path: str = Field("")
+    class_id: int = Field(1)
+    min_score: float = Field(0.25, example=0.30)
+    tensorrt_precision: Optional[int] = Field(None, example=32)
+
+    @validator("device")
+    def validate_device(cls, device):
+        if device not in ["x86", "x86-gpu", "Jetson", "EdgeTPU", "Dummy"]:
+            raise ValueError('Not valid Device. Try one of the following: "x86", "x86-gpu", "Jetson", "EdgeTPU",'
+                             '"Dummy".')
+        return device
+
+    @validator("tensorrt_precision")
+    def validate_tensorrt_precision(cls, tensorrt_precision):
+        if tensorrt_precision is not None:
+            if tensorrt_precision not in [16, 32]:
+                raise ValueError('Not valid tensorrt_precision. Accepted values: 16, 32')
+
+        return tensorrt_precision
+
+    @validator("name")
+    def validate_name(cls, name):
+        if name not in ["openvino", "openpifpaf_tensorrt", "mobilenet_ssd_v2", "openpifpaf", "yolov3",
```

Comment:
This validation is conditional to the device type. Not every one of these is supported for all devices. But I don't know if we want to maintain such a validation. Opinions?

Reply:
Oh I see you added that separately. What is the point of this validator given the `check_models_and_device` function?

Reply:
The truth is that it is enough for validations if I remove `validate_name`. However, this validator will help the user to know what all the options are available for `model_name`.

Reply:
For me there isn't much, it duplicates a validation.

Reply:
Removed!